### ISSUE TITLE
MOTD messages are repeated

### ISSUE THREAD

**Issue by OddBloke** [fwupd/fwupd#1698]
When SSHing into my desktop, I was greeted with _182_ lines of MOTD content from fwupd. This content was repeated 3 times:

```
B450M DS3H
│
└─Unifying Receiver:
  │   Device ID:           cf3685ba249d3d98602047341d6f5a5556a6ac05
  │   Summary:             A miniaturised USB wireless receiver
  │   Current version:     RQR12.07_B0029
  │   Bootloader Version:  BOT01.02_B0014
  │   Vendor:              Logitech, Inc. (USB:0x046D)
  │   Install Duration:    30 seconds
  │   GUIDs:               9d131a0c-a606-580f-8eda-80587250b8d6
  │                        fcf55bf5-767b-51ce-9c17-f6f538c4ee9f
  │                        279ed287-3607-549e-bacc-f873bb9838c4
  │   Device Flags:        • Updatable
  │                        • Supported on remote server
  │ 
  ├─Unifying Receiver (RQR12) Device Update:
  │     New version:       RQR12.10_B0032
  │     Remote ID:         lvfs
  │     Summary:           Firmware for the Logitech Unifying Receiver (RQR12.xx)
  │     License:           Proprietary
  │     Size:              56.8 kB
  │     Vendor:            Logitech
  │     Duration:          30 seconds
  │     Flags:             is-upgrade
  │     Description:       This release addresses an encrypted keystroke injection vulnerability sent by pointing devices. The vulnerability is complex to replicate and would require a hacker to be physically close to a target.
  │     
  │     A few of Logitech's devices used to send select buttons in an unencrypted way, and in an effort to protect against this vulnerability, Logitech removed the feature. Affected hardware is:
  │     
  │      • Wireless Mouse M335
  │      • Zone Touch Mouse T400
  │      • Wireless Mouse M545
  │      • Wireless Mouse M560
  │      • Touch Mouse M600
  │      • Touch Mouse T620
  │      • Wireless Rechargeable Touchpad T650
  │     
  │     Although Logitech does not recommend it, these features may be re-activated by keeping/downgrading the receiver to an older firmware.
  │   
  └─Unifying Receiver (RQR12) Device Update:
        New version:       RQR12.08_B0030
        Remote ID:         lvfs
        Summary:           Firmware for the Logitech Unifying Receiver (RQR12.xx)
        License:           Proprietary
        Size:              72.7 kB
        Vendor:            Logitech
        Duration:          30 seconds
        Flags:             is-upgrade
        Description:       This release addresses an encrypted keystroke injection issue known as Bastille security issue #13. The vulnerability is complex to replicate and would require a hacker to be physically close to a target.
        
        A few of Logitech's devices used to send select buttons in an unencrypted way, and in an effort to protect against this vulnerability, Logitech removed the feature. Affected hardware is:
        
         • Wireless Mouse M335
         • Zone Touch Mouse T400
         • Wireless Mouse M545
         • Wireless Mouse M560
         • Touch Mouse M600
         • Touch Mouse T620
         • Wireless Rechargeable Touchpad T650
        
        Although Logitech does not recommend it, these features may be re-activated by keeping/downgrading the receiver to an older firmware.
```

I've attached [/run/private/motd.d/85-fwupd](https://github.com/fwupd/fwupd/files/4046424/85-fwupd.txt). (You can see that this has the same content _4_ times now, so clearly something is appending when it should be replacing.)

(This was originally reported in Ubuntu at https://bugs.launchpad.net/ubuntu/+source/fwupd/+bug/1859180.)

## Boilerplate

To help us pinpoint your issue, please insert the output of the
following commands when ran on the system with the issue:

```shell
$ fwupdmgr --version
client version:	1.3.6
compile-time dependency versions
	gusb:	0.3.0
	efivar:	37
daemon version:	1.3.6
```

```shell
$ fwupdmgr get-devices
<snip>
├─Unifying Receiver:
│     Device ID:           cf3685ba249d3d98602047341d6f5a5556a6ac05
│     Summary:             A miniaturised USB wireless receiver
│     Current version:     RQR12.07_B0029
│     Bootloader Version:  BOT01.02_B0014
│     Vendor:              Logitech, Inc. (USB:0x046D)
│     Install Duration:    30 seconds
│     GUIDs:               9d131a0c-a606-580f-8eda-80587250b8d6
│                          fcf55bf5-767b-51ce-9c17-f6f538c4ee9f
│                          279ed287-3607-549e-bacc-f873bb9838c4
│     Device Flags:        • Updatable
│                          • Supported on remote server
<snip>
```
(Just included the relevant portion here.)

Please answer the following questions:

- Operating system and version: Ubuntu 20.04 (development release)
- How did you install fwupd (ex: `from source`, `pacman`, `apt-get`, etc): it's in the base system, so apt
- Have you tried rebooting? Don't think it would change matters, as this is intentional behaviour.
- Are you using an NVMe disk? No

**Comment by hughsie** [comment]
@superm1 ?

**Comment by OddBloke** [comment]
As requested in the Ubuntu bug:

```
$ systemctl show fwupd-refresh.service | cat
Type=oneshot
Restart=no
NotifyAccess=none
RestartUSec=100ms
TimeoutStartUSec=infinity
TimeoutStopUSec=1min 30s
TimeoutAbortUSec=1min 30s
RuntimeMaxUSec=infinity
WatchdogUSec=0
WatchdogTimestampMonotonic=0
RootDirectoryStartOnly=no
RemainAfterExit=no
GuessMainPID=yes
SuccessExitStatus=2
MainPID=0
ControlPID=0
FileDescriptorStoreMax=0
NFileDescriptorStore=0
StatusErrno=0
Result=success
ReloadResult=success
CleanResult=success
UID=[not set]
GID=[not set]
NRestarts=0
OOMPolicy=stop
ExecMainStartTimestamp=Fri 2020-01-10 09:30:21 EST
ExecMainStartTimestampMonotonic=169999562244
ExecMainExitTimestamp=Fri 2020-01-10 09:30:21 EST
ExecMainExitTimestampMonotonic=169999651794
ExecMainPID=548705
ExecMainCode=1
ExecMainStatus=0
ExecStart={ path=/usr/bin/fwupdmgr ; argv[]=/usr/bin/fwupdmgr refresh --no-metadata-check ; ignore_errors=no ; start_time=[n/a] ; stop_time=[n/a] ; pid=0 ; code=(null) ; status=0/0 }
ExecStart={ path=/usr/bin/fwupdmgr ; argv[]=/usr/bin/fwupdmgr get-updates --log 85-fwupd ; ignore_errors=no ; start_time=[n/a] ; stop_time=[n/a] ; pid=0 ; code=(null) ; status=0/0 }
ExecStartEx={ path=/usr/bin/fwupdmgr ; argv[]=/usr/bin/fwupdmgr refresh --no-metadata-check ; flags= ; start_time=[n/a] ; stop_time=[n/a] ; pid=0 ; code=(null) ; status=0/0 }
ExecStartEx={ path=/usr/bin/fwupdmgr ; argv[]=/usr/bin/fwupdmgr get-updates --log 85-fwupd ; flags= ; start_time=[n/a] ; stop_time=[n/a] ; pid=0 ; code=(null) ; status=0/0 }
Slice=system.slice
MemoryCurrent=[not set]
CPUUsageNSec=[not set]
EffectiveCPUs=
EffectiveMemoryNodes=
TasksCurrent=[not set]
IPIngressBytes=[no data]
IPIngressPackets=[no data]
IPEgressBytes=[no data]
IPEgressPackets=[no data]
IOReadBytes=18446744073709551615
IOReadOperations=18446744073709551615
IOWriteBytes=18446744073709551615
IOWriteOperations=18446744073709551615
Delegate=no
CPUAccounting=no
CPUWeight=[not set]
StartupCPUWeight=[not set]
CPUShares=[not set]
StartupCPUShares=[not set]
CPUQuotaPerSecUSec=infinity
CPUQuotaPeriodUSec=infinity
AllowedCPUs=
AllowedMemoryNodes=
IOAccounting=no
IOWeight=[not set]
StartupIOWeight=[not set]
BlockIOAccounting=no
BlockIOWeight=[not set]
StartupBlockIOWeight=[not set]
MemoryAccounting=yes
DefaultMemoryLow=0
DefaultMemoryMin=0
MemoryMin=0
MemoryLow=0
MemoryHigh=infinity
MemoryMax=infinity
MemorySwapMax=infinity
MemoryLimit=infinity
DevicePolicy=auto
TasksAccounting=yes
TasksMax=19018
IPAccounting=no
UMask=0022
LimitCPU=infinity
LimitCPUSoft=infinity
LimitFSIZE=infinity
LimitFSIZESoft=infinity
LimitDATA=infinity
LimitDATASoft=infinity
LimitSTACK=infinity
LimitSTACKSoft=8388608
LimitCORE=infinity
LimitCORESoft=0
LimitRSS=infinity
LimitRSSSoft=infinity
LimitNOFILE=524288
LimitNOFILESoft=1024
LimitAS=infinity
LimitASSoft=infinity
LimitNPROC=63395
LimitNPROCSoft=63395
LimitMEMLOCK=65536
LimitMEMLOCKSoft=65536
LimitLOCKS=infinity
LimitLOCKSSoft=infinity
LimitSIGPENDING=63395
LimitSIGPENDINGSoft=63395
LimitMSGQUEUE=819200
LimitMSGQUEUESoft=819200
LimitNICE=0
LimitNICESoft=0
LimitRTPRIO=0
LimitRTPRIOSoft=0
LimitRTTIME=infinity
LimitRTTIMESoft=infinity
OOMScoreAdjust=0
Nice=0
IOSchedulingClass=0
IOSchedulingPriority=0
CPUSchedulingPolicy=0
CPUSchedulingPriority=0
CPUAffinity=
NUMAPolicy=n/a
NUMAMask=
TimerSlackNSec=50000
CPUSchedulingResetOnFork=no
NonBlocking=no
StandardInput=null
StandardInputData=
StandardOutput=journal
StandardError=null
TTYReset=no
TTYVHangup=no
TTYVTDisallocate=no
SyslogPriority=30
SyslogLevelPrefix=yes
SyslogLevel=6
SyslogFacility=3
LogLevelMax=-1
LogRateLimitIntervalUSec=0
LogRateLimitBurst=0
SecureBits=0
CapabilityBoundingSet=cap_chown cap_dac_override cap_dac_read_search cap_fowner cap_fsetid cap_kill cap_setgid cap_setuid cap_setpcap cap_linux_immutable cap_net_bind_service cap_net_broadcast cap_net_admin cap_net_raw cap_ipc_lock cap_ipc_owner cap_sys_rawio cap_sys_chroot cap_sys_ptrace cap_sys_pacct cap_sys_admin cap_sys_boot cap_sys_nice cap_sys_resource cap_sys_time cap_sys_tty_config cap_mknod cap_lease cap_audit_write cap_audit_control cap_setfcap cap_mac_override cap_mac_admin cap_syslog cap_wake_alarm cap_block_suspend cap_audit_read
AmbientCapabilities=
User=fwupd-refresh
Group=fwupd-refresh
DynamicUser=yes
RemoveIPC=yes
MountFlags=
PrivateTmp=yes
PrivateDevices=no
ProtectKernelTunables=no
ProtectKernelModules=yes
ProtectKernelLogs=no
ProtectControlGroups=yes
PrivateNetwork=no
PrivateUsers=no
PrivateMounts=no
ProtectHome=read-only
ProtectSystem=strict
SameProcessGroup=no
UtmpMode=init
IgnoreSIGPIPE=yes
NoNewPrivileges=yes
SystemCallFilter=~chroot fsconfig fsmount fsopen fspick mount move_mount open_tree pivot_root umount umount2
SystemCallErrorNumber=0
LockPersonality=no
RestrictAddressFamilies=AF_INET AF_INET6 AF_NETLINK AF_UNIX
RuntimeDirectoryPreserve=yes
RuntimeDirectoryMode=0755
RuntimeDirectory=motd.d
StateDirectoryMode=0755
CacheDirectoryMode=0755
CacheDirectory=fwupdmgr
LogsDirectoryMode=0755
ConfigurationDirectoryMode=0755
TimeoutCleanUSec=infinity
MemoryDenyWriteExecute=no
RestrictRealtime=yes
RestrictSUIDSGID=yes
RestrictNamespaces=no
MountAPIVFS=no
KeyringMode=private
ProtectHostname=no
KillMode=control-group
KillSignal=15
RestartKillSignal=15
FinalKillSignal=9
SendSIGKILL=yes
SendSIGHUP=no
WatchdogSignal=6
Id=fwupd-refresh.service
Names=fwupd-refresh.service
Requires=-.mount sysinit.target system.slice
Conflicts=shutdown.target
Before=shutdown.target
After=systemd-tmpfiles-setup.service systemd-networkd.service basic.target NetworkManager.service -.mount connman.service sysinit.target network.target fwupd-refresh.timer systemd-journald.socket network-online.target system.slice
TriggeredBy=fwupd-refresh.timer
RequiresMountsFor=/run/motd.d /var/tmp /tmp /var/cache/fwupdmgr
Documentation=man:fwupdmgr(1)
Description=Refresh fwupd metadata and update motd
LoadState=loaded
ActiveState=inactive
SubState=dead
FragmentPath=/lib/systemd/system/fwupd-refresh.service
UnitFileState=static
UnitFilePreset=disabled
StateChangeTimestamp=Fri 2020-01-10 09:30:21 EST
StateChangeTimestampMonotonic=169999651937
InactiveExitTimestamp=Fri 2020-01-10 09:30:19 EST
InactiveExitTimestampMonotonic=169998188380
ActiveEnterTimestampMonotonic=0
ActiveExitTimestampMonotonic=0
InactiveEnterTimestamp=Fri 2020-01-10 09:30:21 EST
InactiveEnterTimestampMonotonic=169999651937
CanStart=yes
CanStop=yes
CanReload=no
CanIsolate=no
CanClean=runtime cache
StopWhenUnneeded=no
RefuseManualStart=no
RefuseManualStop=no
AllowIsolate=no
DefaultDependencies=yes
OnFailureJobMode=replace
IgnoreOnIsolate=no
NeedDaemonReload=no
JobTimeoutUSec=infinity
JobRunningTimeoutUSec=infinity
JobTimeoutAction=none
ConditionResult=yes
AssertResult=yes
ConditionTimestamp=Fri 2020-01-10 09:30:19 EST
ConditionTimestampMonotonic=169998186712
AssertTimestamp=Fri 2020-01-10 09:30:19 EST
AssertTimestampMonotonic=169998186713
Transient=no
Perpetual=no
StartLimitIntervalUSec=10s
StartLimitBurst=5
StartLimitAction=none
FailureAction=none
SuccessAction=none
InvocationID=a1dfabcccca9457f974791697fb1d844
CollectMode=inactive
```

**Comment by OddBloke** [comment]
Oh, and apologies for not reading your request correctly: I haven't enabled `fwupd-refresh.service` manually.

**Comment by superm1** [comment]
Thanks for reporting.  Yes, I can reproduce this.  I think there are 3 distinct things going on that we should address.

1. The output is repeated multiple times.  This appears to be because the default behavior is to append the file: https://github.com/fwupd/fwupd/blob/master/src/fu-util.c#L2795.  I think we should change that to wipe the file by default.

2. The logging doesn't indicate what this is.  For someone logging into a system via SSH seeing a tree output doesn't exactly make sense without context.  We probably at a minimum need a header saying something along the lines of "You have firmware updates available for the following devices".   

3. Even though `fwupd-refresh.service` has the preset set to false, `fwupd-refresh.timer` doesn't.
This was intended to be disabled by default, but something that users can opt into.

```
$ systemctl status fwupd-refresh.timer
● fwupd-refresh.timer - Refresh fwupd metadata regularly
   Loaded: loaded (/lib/systemd/system/fwupd-refresh.timer; enabled; vendor preset: enabled)

$ systemctl status fwupd-refresh.service
● fwupd-refresh.service - Refresh fwupd metadata and update motd
   Loaded: loaded (/lib/systemd/system/fwupd-refresh.service; static; vendor preset: disabled)
```

**Comment by OddBloke** [comment]
I've also filed #1699, as this append behaviour seems like it could be fixed independent of addressing the actual content that's getting appended. :-)

**Comment by superm1** [comment]
This was fixed in #1700
